agent/src/test/java/com/teamscale/jacoco/agent/AgentHttpServerTest.java

```diff
@@ -0,0 +1,84 @@
+package com.teamscale.jacoco.agent;
+
+import com.teamscale.client.TeamscaleServer;
+import com.teamscale.jacoco.agent.options.AgentOptionParseException;
+import com.teamscale.jacoco.agent.options.AgentOptions;
+import com.teamscale.jacoco.agent.options.AgentOptionsParser;
+import com.teamscale.jacoco.agent.upload.UploaderException;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AgentHttpServerTest {
+
+	private Agent agent;
+	private final String defaultCommitMessage = "Some Message";
+
+	/** Starts the http server to control the agent */
+	@BeforeEach
+	public void setup() throws AgentOptionParseException, UploaderException {
+		AgentOptions options = AgentOptionsParser.parse("" +
```

Comment:
As this does not test the `AgentOptionsParser`, I would prefer if it simply constructed the `AgentOptions` directly, perhaps with the help of some `TestAgentOptionsBuilder` (using the Test Data Builder pattern). See for example `TestNamedServerDetailsBuilder` in the Teamscale codebase.

Then, only explicitly set the options in the test that are actually relevant to understand this particular test case.